File: cgi-bin/ruby/env.rb
#!/usr/bin/ruby -w

print "Content-type: text/html \n\n";
print "<!doctype html>";
print "<html><head><title>Environment Var PHP</title></head><body>";
print "<h1>Variables Sent By Server</h1>";
print "<table><tr><th>Variable</th><th>Value</th></tr>";


ENV.sort.each do |key, val|
 if(key.index("REQUEST_") ===nil) 
   print "<tr><td>" + key  + "</td><td>" + val  + "</td></tr>";
 end
end


print "</table>";
print "<h1>Variables Sent By Browser</h1>";
print "<table><tr><th>Variable</th><th>Value</th></tr>";

ENV.sort.each do |key, val|
 if(key.index("REQUEST_") != nil) 
   print "<tr><td>" + key  + "</td><td>" + val  + "</td></tr>";
 end
end
print "</table>"
print "</body>\n</html>";

File: cgi-bin/php/env.php
<?php
$arr = ksort($_SERVER);
$req = array();
header("Content-type: text/html");
print "<!doctype html>";
print "<html><head><title>Environment Var PHP</title></head><body>";
print "<h1>Variables Sent By Server</h1>";
print "<table><tr><th>Variable</th><th>Value</th></tr>";
foreach($_SERVER as $key=>$val)
{
 if(strpos($key,"REQUEST_")===false)
 {
   print "<tr><td>".$key."</td><td>".$val."</td></tr>";
 }
 else
 {
   $req[$key] = $val;
 }
}
print "</table>";
print "<h1>Variables Sent By Browser</h1>";
print "<table><tr><th>Variable</th><th>Value</th></tr>";
foreach($req as $key=>$val)
{
   print "<tr><td>".$key."</td><td>".$val."</td></tr>";
}
print "</table></body></html>";
?>
